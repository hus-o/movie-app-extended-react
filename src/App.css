.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

ul{
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  max-width: 90%;
}

li{width: 25%;}

a{
  text-decoration: none;
  color: white;

}
