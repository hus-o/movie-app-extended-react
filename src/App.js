import React, { Component } from 'react';
import './App.css';
import Search from "./components/search"
import Results from "./components/movieResults"
import Info from "./components/moreInfo"

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      searchResults: [],
      selectedMovie: []
     }

  }

  getSearchResults = (results) => {
    this.setState({searchResults:results})
  }

  getSelectedMovie = (movie) => {
    console.log(this.selectedMovie)
    this.setState({selectedMovie:[movie]})
    console.log(this.selectedMovie)
  }
  
  render() {
    return (
      <div className="App">
        <Search searchResults = {this.getSearchResults} />
        <Results searchResults = {this.state.searchResults} selectedMovie = {this.getSelectedMovie}/>
        <Info selectedMovie = {this.state.selectedMovie}/>
      </div>
    );
  }
}
 

export default App;
