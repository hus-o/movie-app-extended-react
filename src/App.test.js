import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import Search from "./components/Search";
import renderer from 'react-test-renderer';

test('fake test', () => {
  expect(true).toBeTruthy();
});

test('renders correctly', () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it searches', () =>{
  
})