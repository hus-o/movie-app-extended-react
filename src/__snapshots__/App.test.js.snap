// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search for a movie"
    type="text"
    value=""
  />
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
